src/solc_0.8/raffle/PeopleOfCrypto.sol

```diff
@@ -0,0 +1,257 @@
+        } else if (waveType == 2) {
+            require(IERC1155(contractAddress).balanceOf(_wallet, erc1155Id) > 0, "No NFT");
+        }
+
+        uint256 _price = price(_amount);
+        if (_price > 0) {
+            SafeERC20.safeTransferFrom(IERC20(_msgSender()), _wallet, sandOwner, _price);
+        }
+
+        waveOwnerToClaimedCounts[_wallet][indexWave - 1] += _amount;
+
+        waveTotalMinted += _amount;
+
+        for (uint256 i = 0; i < _amount; i++) {
+            uint256 tokenId = getRandomToken(_wallet, totalSupply());
+            _safeMint(_wallet, tokenId);
+        }
+    }
+
+    function checkSignature(
+        address _wallet,
+        uint256 _signatureId,
+        address _contractAddress,
+        uint256 _chainId,
+        bytes memory _signature
+    ) public pure returns (address) {
+        return
+            ECDSA.recover(
+                keccak256(
+                    abi.encodePacked(
+                        "\x19Ethereum Signed Message:\n32",
+                        keccak256(abi.encode(_wallet, _signatureId, _contractAddress, _chainId))
+                    )
+                ),
+                _signature
+            );
+    }
+
+    function checkPersonalizationSignature(
+        uint256 _signatureId,
+        address _contractAddress,
+        uint256 _chainId,
+        bytes memory _signature,
+        uint256 _tokenId,
+        uint256 _personalizationMask
+    ) public pure returns (address) {
+        return
+            ECDSA.recover(
+                keccak256(
+                    abi.encodePacked(
+                        "\x19Ethereum Signed Message:\n32",
+                        keccak256(abi.encode(_signatureId, _contractAddress, _chainId, _tokenId, _personalizationMask))
+                    )
+                ),
+                _signature
+            );
+    }
+
+    // Thx Cyberkongs VX <3
+    function getRandomToken(address _wallet, uint256 _totalMinted) private returns (uint256) {
+        uint256 remaining = MAX_SUPPLY - _totalMinted;
+        uint256 rand =
+            uint256(keccak256(abi.encodePacked(_wallet, block.difficulty, block.timestamp, remaining))) % remaining;
+        uint256 value = rand;
+
+        if (availableIds[rand] != 0) {
+            value = availableIds[rand];
+        }
+
+        if (availableIds[remaining - 1] == 0) {
+            availableIds[rand] = remaining - 1;
+        } else {
+            availableIds[rand] = availableIds[remaining - 1];
+        }
+
+        return value;
+    }
+
+    function toggleSale() external onlyOwner {
+        paused = paused == 0 ? 1 : 0;
+        emit TogglePaused(paused == 1);
+    }
+
+    function personalize(
+        uint256 _signatureId,
+        bytes memory _signature,
+        uint256 _tokenId,
+        uint256 _personalizationMask
+    ) external {
+        require(_msgSender() != allowedToExecutePersonalization, "Not allowed");
```

Comment:
not sure of the purpose of allowedToExecutePersonalization, I think we should remove the whole line